hal/inc/Keypad.h

```diff
@@ -0,0 +1,43 @@
+#ifndef _KEYPAD_H_
+#define _KEYPAD_H_
+
+#define KEYPAD_DEBOUNCING_TIME 20
+#define COL_SIZE               4
+#define ROW_SIZE               4
```

Comment:
Actually we have multiple types of keypads. I think it would be better if we have a templated class that introduce multiple types of keypad. e.g.
```
Keypad<4, 4> kpd; // represent 4x4 keypad
Keypad<2, 3> kpd; // represent 2x3 keypad
```
And if you think there will be totally separate algorithm for each type you can use template specialization.